src/plugins/data/common/search/strategies/sql_search/types.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type {
+  SqlGetAsyncRequest,
+  SqlQueryRequest,
+  SqlQueryResponse,
+} from '@elastic/elasticsearch/lib/api/types';
+import { IKibanaSearchRequest, IKibanaSearchResponse } from '../../types';
+
+export const SQL_SEARCH_STRATEGY = 'sql';
+
+export type SqlRequestParams =
+  | Omit<SqlQueryRequest, 'wait_for_completion_timeout' | 'keep_alive' | 'keep_on_completion'>
```

Comment:
Does this mean we wouldn't allow overriding the value for `wait_for_completion_timeout` by consumers?

Reply:
Good point. I agree that allowing `wait_for_completion_timeout` wait for completion makes sense since it doesn't interfere with search sessions. Fixing